cola/linalg/inverse/gmres.py

```diff
@@ -1,10 +1,31 @@
-from cola.ops import LinearOperator
-from cola.ops import Array
+from dataclasses import dataclass
+
+from cola.linalg.algorithm_base import Algorithm
 from cola.linalg.decompositions.arnoldi import arnoldi
+from cola.ops import Array, LinearOperator
 from cola.utils import export
 from cola.utils.custom_autodiff import iterative_autograd
-from cola.linalg.algorithm_base import Algorithm
-from dataclasses import dataclass
+
+
+@dataclass
+class LSTSQ_GMRES(Algorithm):
```

Comment:
why new algorithm, let's consolidate and just have the algorithm used in the Pinv method

Reply:
Based on our SVD discussion it makes sense to have it on the dispatch rule.

Reply:
The only complication that I'm finding is that I need to modify the RHS b as `A.H @ b`. It is not clear to me that I can do so without creating a new algorithm class. Wdyt?